### PR TITLE
Carets - Guillermina and Marisa

### PR DESCRIPTION
**Pull request by murog**
# Scrabble
Congratulations! You're submitting your assignment.

## Comprehension Questions

|  Feature 	|   Feedback	|
|---	|---	|
|   How do you feel you and your partner did in sharing responsibilities?	| We benefited from using a timer and being conscious of the driver/navigator roles.   	|
|   Why do you think we chose to make the `score` method in the `Scoring` class a **class method** instead of an **instance method**? |  Instance methods can only be used on an instance of the class.  By implementing it as a class method, we were able to feed it an argument and have it compute a score without creating an instance of a "Scoring object."   |
|   Describe an example of a test you wrote for a _nominal case_. |   A nominal case we tested for was that the play method in the Player class should return a score for the input word.  So if player played the word "aaa" it should return 3.  |
|   Describe an example of a test you wrote for an _edge case_. |   Our scoring class does not score numbers, so play should not work with strings that contain numbers.  So, if "friends4ever" was attempted to be used as an argument for the play method, it should raise an ArgumentError. |
|   What was your final test coverage percentage?	|   100% 	|
|   Is there a specific piece of code you'd like feedback on?	|   We would like feedback on whether we tested for appropriate edge cases.	|

### CODE REVIEW

**Comment by tildeee** [comment]
# Scrabble
## What We're Looking For

|  Feature  |   Feedback |
|---	|---	|
|  **General**	|   	|
|   Answered comprehension questions  |  x  |
|   Both Teammates contributed to the codebase	|  x  |
|   Regular Commits with meaningful commit messages  |  x  |
|  **Scoring class**	|   	|
|  `score` method	|  x  |
|   Uses appropriate data structure to store the letter score	|  x  |
|   Appropriately handles edge cases	|  x  |
|   Tests `score` edge cases (one-letter word, 7 letter word)	|  x  |
|  `highest_scoring_word` method	|  x  |
|   Appropriately handles edge cases	|  x  |
|   Tests `highest_scoring_word` edge cases (ties, no words, no array)	|  x  |
|  **Player class**	|   	|
|  Uses an array to keep track of words played	|  x  |
|  Uses existing code to create `highest_scoring_word` and `highest_word_score`	|  used `Scoring.score`, not `Scoring.highest_score_from`  |
|  Returns `true` or `false` from `won?` method |  x  |
|  Tests edge cases on the `play(word)` method for words that cannot be scored |   just checks for numbers, not symbols or empty   |
|  **TileBag class**	|   	|
|  Uses a data structure to keep the set of default tiles in the bag	|  x  |
|  `draw_tiles` method uses the parameter to draw a set of random tiles |  x  |
| `tiles_remaining` method returns a count	|   x	|
|  Tests `draw_tiles` edge cases for different parameter inputs	|   	|

File: lib/tilebag.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        @bag.delete_at(0)
+      end
+      update_tiles
+      return grab
+    end
+
+    private
+    def update_tiles
+      @tiles_remaining = @bag.length
+    end
```
Comment (tildeee) [inline lib/tilebag.rb:56]: This code requires you to remember to call `update_tiles` after every action in order to update `@tiles_remaining`

may want to refactor this to
```ruby
def tiles_remaining
  return @bag.length
end
```

File: lib/scrabble.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          score += 4
+        elsif VALUES[5].include? letter.upcase
+          score += 5
+        elsif VALUES[8].include? letter.upcase
+          score += 8
+        elsif VALUES[10].include? letter.upcase
+          score += 10
+        else
+          raise ArgumentError.new "Invalid letter in word"
+        end
```
Comment (tildeee) [inline lib/scrabble.rb:45]: What if values changed and you updated the `VALUES` data to include ` 9 => ["XXX"]`? Then you would need to update this logic in order to have a new line `elsif VALUES[9].include? ...` etc. There's opportunity here to refactor this so that you don't need to have all of these conditionals.

File: lib/player.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      @plays << input_word
+      total_score
+
+      return score
+    end
+
+    def total_score
+      return 0 if @plays.length == 0
+      scores = @plays.map{ |word| Scrabble::Scoring.score(word) }
+      @total = scores.inject(:+)
```
Comment (tildeee) [inline lib/player.rb:33]: instead of recalculating the scores of each word in plays, there may be opportunity to refactor this

File: lib/scrabble.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    def self.tie_breaker(tie_array)
+      seven_letters = tie_array.select{|word| word.length == 7}
+      if seven_letters.length >= 1
+        return seven_letters[0] # => winning word, only 1 word with 7 letters
+      end
+      shortest_word = tie_array.inject do |memo, word|
+        memo.length <= word.length ? memo : word
+      end
+      if shortest_word.length >= 1
```
Comment (tildeee) [inline lib/scrabble.rb:82]: this conditional may be unnecessary?

**Comment by tildeee** [comment]
Nice work. In some places, you turned things into instance variables that required being updated, which might have been easier if they had been methods.
